src/online2/online-endpoint.cc

```diff
@@ -137,11 +137,20 @@ bool EndpointDetected<LatticeFasterOnlineDecoderTpl<fst::Fst<fst::StdArc> > >(
 
 
 template
-bool EndpointDetected<LatticeFasterOnlineDecoderTpl<fst::GrammarFst> >(
```

Comment:
There are lots of changes where I think it would be simpler to just rely on a
`using GrammarFst = GrammarFstTpl<const fst::ConstFst<fst::StdArc> >` in the grammar-fst.h header.

Reply:
or maybe ConstGrammarFst and VectorGrammarFst.